data-hibernate-reactive/src/main/java/io/micronaut/data/hibernate/reactive/operations/DefaultHibernateReactiveRepositoryOperations.java

```diff
@@ -126,29 +125,25 @@ protected void setParameter(Stage.Query<?> query, String parameterName, Object v
     @Override
     protected void setParameter(Stage.Query<?> query, String parameterName, Object value, Argument argument) {
         if (value == null) {
-            ParameterExpression parameter = sessionFactory.getCriteriaBuilder().parameter(argument.getType(), parameterName);
-            query.setParameter(parameter, null);
+            query.setParameter(parameterName, null, argument.getType());
         } else {
             query.setParameter(parameterName, value);
         }
     }
 
     @Override
     protected void setParameterList(Stage.Query<?> query, String parameterName, Collection<Object> value) {
-        query.setParameter(parameterName, value);
+        query.setParameterList(parameterName, value);
     }
 
     @Override
     protected void setParameterList(Stage.Query<?> query, String parameterName, Collection<Object> value, Argument argument) {
-        //QueryParameterBindings
-        ParameterExpression parameter = sessionFactory.getCriteriaBuilder().parameter(argument.getType(), parameterName);
-        query.setParameter(parameter, value);
+        query.setParameterList(parameterName, value, argument.getType());
```

Comment:
This fixed some tests but setting null parameter doesn't seem to work properly, seems like binding is not working properly or something.

Reply:
Yes, I have added back `void "test IN queries"()` that works now. But those with null parameter are still not working properly, cannot dig deep enough as some transaction interrupted error is thrown. Probably it's not able to bind null values or something, although it should as it was told what type it is etc.